### PR TITLE
dev/core#3463 - CiviMail - prefer frontend_title from group for maili…

### PR DESCRIPTION
**Pull request by ufundo**
…ng.group token

Overview
----------------------------------------
Implementation for https://lab.civicrm.org/dev/core/-/issues/3463

Before
----------------------------------------
{mailing.group} token in CiviMail always uses the `title` field for each of the groups in the mailing.

After
----------------------------------------
Uses the `frontend_title` field for any groups in the mailing where this is set.

Technical Details
----------------------------------------
I was slightly worried about issues with permissions / bootstrapping. I've test sending scheduled mailings with a cron user with the sysadmin-docs-recommended cron permissions of View all contacts, Access CiviCRM, Access CiviMail and seems to work ok :)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@ufundo this makes sense to be but you need to check the style warnings https://test.civicrm.org/job/CiviCRM-Core-PR/49060/checkstyle/

@seamuslee001 @agileware-justin

File: CRM/Mailing/BAO/Mailing.php [hunk 1]
```diff
@@ -1419,22 +1419,26 @@ public function &getGroupNames() {
...
-                        AND     $mgtable.entity_table = '$group'
-                        AND     $mgtable.group_type = 'Include'
-                    ORDER BY    $group.name");
+    $mailingGroups = \Civi\Api4\MailingGroup::get()
+      ->addSelect('group.title', 'group.frontend_title')
+      ->addJoin('Group AS group', 'LEFT', ['entity_id', '=', 'group.id'])
+      ->addWhere('mailing_id', '=', $this->id)
+      ->addWhere('entity_table', '=', 'civicrm_group')
+      ->addWhere('group_type', '=', 'Include')
+      ->setLimit(25)
```
Comment (agileware-justin) [inline CRM/Mailing/BAO/Mailing.php:1429]: I would remove this limit 25

**Review by agileware-justin — CHANGES_REQUESTED** [review]

**Comment by agileware-justin** [comment]
@ufundo I agree with this enhancement. The 25 limit looks like it snuck in and suggested this should be removed. I have minor PTSD from this option.

File: CRM/Mailing/BAO/Mailing.php (hunk 1 above)
Comment (ufundo) [reply CRM/Mailing/BAO/Mailing.php]: I left it in thinking 'if you have more than 25 groups in your mailing, this token is going to be a mess' but yes that was probably not a good thought! Have removed now.

**Comment by eileenmcnaughton** [comment]
@agileware-justin - shall I give this merge-on-pass now?

**Comment by agileware-justin** [comment]
@eileenmcnaughton let me test this on DEV and confirm OK.

**Comment by eileenmcnaughton** [comment]
Thanks @agileware-justin !

**Comment by agileware-justin** [comment]
# Existing group token output

Internal group title is output for the {mailing.group} token

image

image

**Comment by agileware-justin** [comment]
# Patch applied, group token output

With the patch applied, public group title is output, if set for the {mailing.group} token. Otherwise will output the group title.



image

image

**Comment by agileware-justin** [comment]
@eileenmcnaughton that's a pass from me.

**Comment by eileenmcnaughton** [comment]
@agileware-justin thanks - I was too mesmerised by your screenshots to be able to figure out if they were right or not :-)

Nice to see some legacy code pattern removed too

**Comment by agileware-justin** [comment]
Screenshot updated.

Agileware ref: CIVICRM-1990

**Comment by eileenmcnaughton** [comment]
lol!

**Pull request merged**